### PR TITLE
Add reformating for parentheses

### PR DESCRIPTION
**Pull request by cassiersg**
I'm quite new to Rust, so I picked this for practice. Please tell me if it's not idiomatic for Rust.

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -125,6 +125,23 @@ impl<'a> FmtVisitor<'a> {
...
     }
 
+    fn rewrite_paren(&mut self, subexpr: &ast::Expr, width: usize, offset: usize) -> String {
+        debug!("rewrite_paren, width: {}, offset: {}", width, offset);
+        // 1 is for opening paren
+        let subexpr_str = self.rewrite_expr(subexpr, width-1, offset+1);
+        debug!("rewrite_paren, subexpr_str: `{}`", subexpr_str);
+        let mut lines = subexpr_str.rsplit('\n');
+        let last_line_len = lines.next().unwrap().len();
+        let last_line_offset = if lines.next().is_none() {offset+1} else {0};
```
Comment (nrc) [inline src/expr.rs]: better to spread this `if` expression over multiple lines

Comment (nrc) [inline src/expr.rs]: and actually better to use a match expression than `if ...is_none()`

File: src/expr.rs [hunk 2]
```diff
@@ -125,6 +125,23 @@ impl<'a> FmtVisitor<'a> {
...
+        if width + offset - last_line_offset - last_line_len > 0 {
+            format!("({})", subexpr_str)
+        } else {
+            // FIXME That's correct unless we have width < 2. Return an Optrion for such cases ?
```
Comment (nrc) [inline src/expr.rs]: typo - Optrion

**Comment by nrc** [comment]
Thanks for the PR! From an idiomatic Rust point of view this looks great, the only non-idomatic thing I noticed was using `if ...is_none()...` rather than `match ...`.

I think I would format so there is never a dangling `)`. I.e., you'd always have `(a + b + c)` or

```
(a + b +
 c + d)
```
never
```
(a + b + c
 )
```

If we ever need that one char to fit in the width, it would be better to either modify the sub-expression or to modify the enclosing expression.

**Comment by cassiersg** [comment]
This last commit avoid dangling `)`, by modifying the sub-expression.

**Comment by cassiersg** [comment]
Thank you for the prompt review.

**Comment by nrc** [comment]
Sweet, thanks

**Pull request merged**